### PR TITLE
fix: Admin email not working

### LINKED ISSUE #12827: [Bug]:  The option to add admin emails does not grant admin access for form login users
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

 The option to add admin emails does not grant admin access for form login users 
image
image

### Steps To Reproduce

1. In the admin setting page
2. under general tab
3. In Admin Email when user adds any email address of the existing user
4. Now login with the existing user email
5. check for admin setting  acess
6. The user doesn't have a admin access

### Public Sample App

_No response_

### Version

release

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
When a new email of an existing user is added to admin emails from the admin settings page, that user does not get admin access instantly. Same happens when an existing user's email is removed from admin settings - permission is not revoked instantly.

Fixes #12827

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java [hunk 1]
```diff
@@ -399,10 +391,7 @@ private Mono<Void> updateAdminUserPolicies(String oldAdminEmailsCsv) {
          * we need to run these two flux immediately because server will be restarted and these changes
          * should be persisted to DB before that
          */
-        return Mono.whenDelayError(
-                removedUserFlux.then(),
-                newUsersFlux.then()
-        );
+        return Flux.merge(removedUserFlux, newUsersFlux);
```
[the remark this answers is not in the document]
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java]: I've updated the PR. BTW not sure about the prefetch parameter, passed total number of emails as the value.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java [hunk 2]
```diff
@@ -232,7 +230,9 @@ public Mono<EnvChangesResponseDTO> applyChanges(Map<String, String> changes) {
                     if (changesCopy.containsKey(APPSMITH_ADMIN_EMAILS.name())) {
                         commonConfig.setAdminEmails(changesCopy.remove(APPSMITH_ADMIN_EMAILS.name()));
                         String oldAdminEmailsCsv = originalValues.get(APPSMITH_ADMIN_EMAILS.name());
-                        dependentTasks = dependentTasks.then(updateAdminUserPolicies(oldAdminEmailsCsv));
+                        dependentTasks = dependentTasks.then(
+                                updateAdminUserPolicies(oldAdminEmailsCsv).collectList().then()
```
[the remark this answers is not in the document]
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java]: Yeah, `collectList` is not required. Updated.

**Review by sharat87 — APPROVED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ce/EnvManagerCEImpl.java]: Yeah sounds good.

**Pull request merged**